samtranslator/model/api/api_generator.py

```diff
@@ -576,27 +596,45 @@ def _construct_api_domain(  # noqa: too-many-branches
 
             record_set_group.RecordSets += self._construct_record_sets_for_domain(self.domain, api_domain_name, route53)
 
-        return domain, basepath_resource_list, record_set_group
+        return ApiDomainResponse(domain, basepath_resource_list, record_set_group)
+
+    def _construct_single_record_set_group(
+        self, domain: Dict[str, Any], api_domain_name: str, route53: Any
+    ) -> Route53RecordSetGroup:
+        hostedZoneId = route53.get("HostedZoneId")
+        hostedZoneName = route53.get("HostedZoneName")
+        domainName = domain.get("DomainName")
+        logical_id_suffix = LogicalIdGenerator("", [hostedZoneId or hostedZoneName, domainName]).gen()
+        logical_id = "RecordSetGroup" + logical_id_suffix
```

Comment:
Would this be identical?


```suggestion
        logical_id = LogicalIdGenerator("RecordSetGroup", [hostedZoneId or hostedZoneName, domainName]).gen()
```